fix: [ML] Anomaly Detection: Advanced job wizard code editors missing title from announcement

alexwizp opened the pull request:
Closes: #216767

**Description**
Dialog modal, flyout, field visible title should be announced for the users, especially using assistive technology to know what dialog modal, flyout opened, what field is active and what is needed to enter in it.

**Changes Made:**

1. Added the required ARIA attributes.
2. Moved the title one level up to correctly set the id.


<!--ONMERGE {"backportTargets":["9.1"]} ONMERGE-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

elasticmachine commented:
Pinging @elastic/kibana-accessibility (Project:Accessibility)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/326534)
* Commit: 55e0e1193c200d777b4411d2c6d098bab0dc9729

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/326534#01987a8e-1d2c-4463-bf27-3120eb6990d3)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326534","number":326534,"commit":"55e0e1193c200d777b4411d2c6d098bab0dc9729"}],"number":326534}
buildkite-pr-comment-->

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

alexwizp committed:
fix: [ML] Anomaly Detection: Advanced job wizard code editors missing title from announcement

# Conflicts:
#	x-pack/platform/plugins/shared/ml/public/application/jobs/new_job/pages/components/common/json_editor_flyout/json_editor_flyout.tsx

alexwizp committed:
Merge branch 'main' into #216767

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/327168)
* Commit: 736485d6b8283f6ddf8ed512065d711d18f660f0

### Failed CI Steps
* [FTR Configs #3](https://buildkite.com/elastic/kibana-pull-request/builds/327168#01988082-98a0-442c-aba7-49de6d80be80)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/5fa1f8a511ea7963313eeacd4167604c2c8bff0a) | [after](https://github.com/elastic/kibana/commit/736485d6b8283f6ddf8ed512065d711d18f660f0) | diff |
| --- | --- | --- | --- |
| `ml` | 5.3MB | 5.3MB | +117.0B |



### History
* :green_heart: [Build #326555](https://buildkite.com/elastic/kibana-pull-request/builds/326555) succeeded 9927c45b6627c17208129761871028c2069c736a
* :broken_heart: [Build #326542](https://buildkite.com/elastic/kibana-pull-request/builds/326542) failed 23a59296938f84a2320bec20f48aa9275cbafd9f
* :broken_heart: [Build #326534](https://buildkite.com/elastic/kibana-pull-request/builds/326534) failed 55e0e1193c200d777b4411d2c6d098bab0dc9729

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327168","number":327168,"commit":"736485d6b8283f6ddf8ed512065d711d18f660f0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326555","number":326555,"commit":"9927c45b6627c17208129761871028c2069c736a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326542","number":326542,"commit":"23a59296938f84a2320bec20f48aa9275cbafd9f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326534","number":326534,"commit":"55e0e1193c200d777b4411d2c6d098bab0dc9729"}],"number":327168}
buildkite-pr-comment-->

alexwizp committed:
Merge branch 'main' into #216767

jgowdyelastic reviewed (approved):
LGTM

peteharverson reviewed (approved):
Tested latest changes and LGTM

kibanamachine commented:
Starting backport for target branches: 9.1

https://github.com/elastic/kibana/actions/runs/16801380302

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/327325)
* Commit: 11130c426b04ff66bf8509c207a8bd32f511c233

### Failed CI Steps
* [FTR Configs #62](https://buildkite.com/elastic/kibana-pull-request/builds/327325#0198839c-0ac4-4c62-b4af-def612b52404)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/327325#0198839c-0ac4-4c62-b4af-def612b52404) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/327325/jobs/0198839c-0ac4-4c62-b4af-def612b52404/artifacts/019883c5-f76e-4fc3-a6bb-de009838c50b) FTR Configs #<span></span>62 / Entity Analytics - Risk Engine @<span></span>ess @<span></span>serverless @<span></span>serverlessQA Risk Scoring Task Execution with auditbeat data with some alerts containing hosts initializing the risk engine @<span></span>skipInServerlessMKI starts the latest transform

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/6bbf934fb65e2e433f0108d7e0028c924c59c641) | [after](https://github.com/elastic/kibana/commit/11130c426b04ff66bf8509c207a8bd32f511c233) | diff |
| --- | --- | --- | --- |
| `ml` | 5.3MB | 5.3MB | +117.0B |



### History
* :yellow_heart: [Build #327168](https://buildkite.com/elastic/kibana-pull-request/builds/327168) was flaky 736485d6b8283f6ddf8ed512065d711d18f660f0
* :green_heart: [Build #326555](https://buildkite.com/elastic/kibana-pull-request/builds/326555) succeeded 9927c45b6627c17208129761871028c2069c736a
* :broken_heart: [Build #326542](https://buildkite.com/elastic/kibana-pull-request/builds/326542) failed 23a59296938f84a2320bec20f48aa9275cbafd9f
* :broken_heart: [Build #326534](https://buildkite.com/elastic/kibana-pull-request/builds/326534) failed 55e0e1193c200d777b4411d2c6d098bab0dc9729

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327325","number":327325,"commit":"11130c426b04ff66bf8509c207a8bd32f511c233"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/327168","number":327168,"commit":"736485d6b8283f6ddf8ed512065d711d18f660f0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326555","number":326555,"commit":"9927c45b6627c17208129761871028c2069c736a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326542","number":326542,"commit":"23a59296938f84a2320bec20f48aa9275cbafd9f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326534","number":326534,"commit":"55e0e1193c200d777b4411d2c6d098bab0dc9729"}],"number":327325}
buildkite-pr-comment-->
